File: frontend/src/app/core/services/usuario.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Usuario } from '../../models';

/*
 * Servicio para la administración de usuarios.
 * Se usa en el módulo admin para gestionar los usuarios del sistema.
 * Se separa de AuthService porque AuthService maneja login/registro,
 * mientras que este servicio maneja el CRUD administrativo.
 */
@Injectable({
  providedIn: 'root'
})
export class UsuarioService {
  private readonly API_URL = '/api/usuarios';

  constructor(private http: HttpClient) {}

  listarTodos(): Observable<Usuario[]> {
    return this.http.get<Usuario[]>(this.API_URL);
  }

  obtenerPorId(id: number): Observable<Usuario> {
    return this.http.get<Usuario>(`${this.API_URL}/${id}`);
  }

  actualizar(id: number, usuario: Usuario): Observable<Usuario> {
    return this.http.put<Usuario>(`${this.API_URL}/${id}`, usuario);
  }

  eliminar(id: number): Observable<any> {
    return this.http.delete(`${this.API_URL}/${id}`);
  }
}

File: SysSupplementsGym/src/main/java/com/syssupplements/gym/service/UsuarioService.java
package com.syssupplements.gym.service;

import com.syssupplements.gym.model.seguridad.Persona;
import com.syssupplements.gym.model.seguridad.Usuario;
import com.syssupplements.gym.repository.PersonaRepository;
import com.syssupplements.gym.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

/*
 * Service para la gestión administrativa de usuarios.
 * Se separa de AuthService porque AuthService maneja autenticación
 * y registro, mientras que este Service maneja la administración
 * de usuarios (CRUD completo para el módulo admin).
 */
@Service
@RequiredArgsConstructor
public class UsuarioService {

    private final UserRepository userRepository;
    private final PersonaRepository personaRepository;

    public Usuario guardar(Usuario usuario) {
        return userRepository.save(usuario);
    }

    public Usuario obtenerPorId(Integer id) {
        return userRepository.findById(id).orElse(null);
    }

    public List<Usuario> listarTodos() {
        return userRepository.findAll();
    }

    public Usuario actualizar(Usuario usuario) {
        return userRepository.save(usuario);
    }

    @Transactional
    public void eliminar(Integer id) {
        Usuario usuario = userRepository.findById(id).orElse(null);
        if (usuario == null) return;

        Persona persona = usuario.getPersona();
        if (persona != null) {
            persona.setActivo(false);
            personaRepository.save(persona);
        }

        usuario.setPersona(null);
        userRepository.delete(usuario);
    }
}
